Source/WebCore/dom/CustomElementReactionQueue.cpp

```diff
@@ -42,84 +42,73 @@
-
-    CustomElementReactionQueueItem(Type type, Payload payload = std::nullopt)
-        : m_type(type)
-        , m_payload(payload)
-    { }
-
-    Type type() const { return m_type; }
-
-    void invoke(Element& element, JSCustomElementInterface& elementInterface)
-    {
-        switch (m_type) {
-        case Type::ElementUpgrade:
-            ASSERT(!m_payload.has_value());
-            elementInterface.upgradeElement(element);
-            break;
-        case Type::Connected:
-            ASSERT(!m_payload.has_value());
-            elementInterface.invokeConnectedCallback(element);
-            break;
-        case Type::Disconnected:
-            ASSERT(!m_payload.has_value());
-            elementInterface.invokeDisconnectedCallback(element);
-            break;
-        case Type::Adopted: {
-            ASSERT(m_payload.has_value() && std::holds_alternative<AdoptedPayload>(m_payload.value()));
-            auto& payload = std::get<AdoptedPayload>(m_payload.value());
-            elementInterface.invokeAdoptedCallback(element, std::get<0>(payload), std::get<1>(payload));
-            break;
-        }
-        case Type::AttributeChanged: {
-            ASSERT(m_payload.has_value() && std::holds_alternative<AttributeChangedPayload>(m_payload.value()));
-            auto& payload = std::get<AttributeChangedPayload>(m_payload.value());
-            elementInterface.invokeAttributeChangedCallback(element, std::get<0>(payload), std::get<1>(payload), std::get<2>(payload));
-            break;
-        }
-        case Type::FormAssociated:
-            ASSERT(m_payload.has_value() && std::holds_alternative<FormAssociatedPayload>(m_payload.value()));
-            elementInterface.invokeFormAssociatedCallback(element, std::get<FormAssociatedPayload>(m_payload.value()).get());
-            break;
-        case Type::FormReset:
-            ASSERT(!m_payload.has_value());
-            elementInterface.invokeFormResetCallback(element);
-            break;
-        case Type::FormDisabled:
-            ASSERT(m_payload.has_value() && std::holds_alternative<FormDisabledPayload>(m_payload.value()));
-            elementInterface.invokeFormDisabledCallback(element, std::get<FormDisabledPayload>(m_payload.value()));
-            break;
-        case Type::FormStateRestore:
-            ASSERT(m_payload.has_value() && std::holds_alternative<FormStateRestorePayload>(m_payload.value()));
-            elementInterface.invokeFormStateRestoreCallback(element, std::get<FormStateRestorePayload>(m_payload.value()));
-            break;
-        }
-    }
+inline CustomElementReactionQueueItem::AdoptedPayload::~AdoptedPayload() = default;
+inline CustomElementReactionQueueItem::FormAssociatedPayload::~FormAssociatedPayload() = default;
+
+inline CustomElementReactionQueueItem::CustomElementReactionQueueItem()
+    : m_type(Type::Invalid)
+{ }
+
+inline CustomElementReactionQueueItem::CustomElementReactionQueueItem(CustomElementReactionQueueItem&& other)
```

Comment:
Could we `= default;` here? Your implementation doesn't look different from what the compiler would generate for us.